src/components/ec/cuda/kernel/ec_cuda_reduce_ops.h

```diff
@@ -53,114 +60,173 @@ cuFloatComplex operator* (const cuFloatComplex & first,
+                _Pragma("unroll") for (i = 0; i < UNROLL; i++)                    \
+                {                                                                 \
+                    tmp1[i] = _OP(tmp1[i], tmp2[i]);                              \
+                }                                                                 \
+            }                                                                     \
+            if (flags & UCC_EEE_TASK_FLAG_REDUCE_WITH_ALPHA) {                    \
+                _Pragma("unroll") for (i = 0; i < UNROLL; i++)                    \
+                {                                                                 \
+                    tmp1[i] = tmp1[i] * (_AlphaType)task.alpha;                   \
+                }                                                                 \
+            }                                                                     \
+            _Pragma("unroll") for (i = 0; i < UNROLL; i++)                        \
+            {                                                                     \
+                d[line + WARP_SIZE * i] = tmp1[i];                                \
+            }                                                                     \
+        }                                                                         \
+        for (i = triggered                                                        \
+                     ? num_lines + threadIdx.x                                    \
+                     : num_lines + threadIdx.x + blockIdx.x * blockDim.x;         \
+             i < count;                                                           \
+             i += triggered ? blockDim.x : blockDim.x * gridDim.x) {              \
+            d[i] = _OP(s[0][i], s[1][i]);                                         \
+            for (j = 2; j < UCC_EE_EXECUTOR_NUM_BUFS; j++) {                      \
+                if (j >= n_srcs) {                                                \
+                    break;                                                        \
+                }                                                                 \
+                d[i] = _OP(d[i], s[j][i]);                                        \
+            }                                                                     \
+        }                                                                         \
+        if (flags & UCC_EEE_TASK_FLAG_REDUCE_WITH_ALPHA) {                        \
+            for (i = triggered                                                    \
+                         ? num_lines + threadIdx.x                                \
+                         : num_lines + threadIdx.x + blockIdx.x * blockDim.x;     \
+                 i < count;                                                       \
+                 i += triggered ? blockDim.x : blockDim.x * gridDim.x) {          \
```

Comment:
i assume this is optimized by compiler anyways, but i personally would move "triggered ? blockDim.x : blockDim.x * gridDim.x" to the local vars.

Reply:
alright. I store it in the variable `step`. I also defined the variable `start` that stores
```
triggered? num_lines + threadIdx.x  : num_lines + threadIdx.x + blockIdx.x * blockDim.x;  
```